ccl/partitionccl: re-enable Test{InitialPartitioning,Repartitioning}

petermattis opened the pull request:
Avoid creating test cases with an `ARRAY` column as the primary
key. Unify the handling of non-indexable column types for test case
generation (`ARRAY`, `JSONB` and `TUPLE`).

Adjust `verifyScansOnNode` to skip `read completed` trace lines for the
system config range. Reads can be performed on this range to renew the
table lease and we want to ignore those reads for the purposes of the
test.

Fixes #28786
Fixes #28789

Release note: None

petermattis committed:
ccl/partitionccl: re-enable Test{InitialPartitioning,Repartitioning}

Avoid creating test cases with an `ARRAY` column as the primary
key. Unify the handling of non-indexable column types for test case
generation (`ARRAY`, `JSONB` and `TUPLE`).

Adjust `verifyScansOnNode` to skip `read completed` trace lines for the
system config range. Reads can be performed on this range to renew the
table lease and we want to ignore those reads for the purposes of the
test.

Fixes #28786
Fixes #28789

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/33409)
<!-- Reviewable:end -->

benesch reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> thank you for finally tracking this down!

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33409#-:-LV22-ThDqrw9nM4k2XO:bgu42yw)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/ccl/partitionccl/partition_test.go, line 849 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/33409#-LV210wg0kEnn-pSoi8p:-LV210wg0kEnn-pSoi8q:b-kyyni7) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd2feeefc4128ff60b5eb09e8fde1e27ebe9d3ec/pkg/ccl/partitionccl/partition_test.go#L849)):*
> ```Go
> 	const schemaFmt = `CREATE TABLE %%s (a %s PRIMARY KEY) PARTITION BY LIST (a) (PARTITION p VALUES IN (%s))`
> 	for semTypeID, semTypeName := range sqlbase.ColumnType_SemanticType_name {
> 		switch sqlbase.ColumnType_SemanticType(semTypeID) {
> ```

very optional nit: consider pulling out a `semType` variable since this same conversion is done a few lines below.
___
*[pkg/ccl/partitionccl/partition_test.go, line 1092 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/33409#-LV21vE98FX2ehVUO7i2:-LV21vE98FX2ehVUO7i3:b9p1f9p) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd2feeefc4128ff60b5eb09e8fde1e27ebe9d3ec/pkg/ccl/partitionccl/partition_test.go#L1092)):*
> ```Go
> 				continue
> 			}
> 			fmt.Printf("trace (%s): %s\n", node, traceLine.String)
> ```

Stray debugging?


<!-- Sent from Reviewable.io -->

petermattis commented:
Done and done (reviewable is not working for me).

petermattis commented:
Hmm, the CI build failed because something is running `make stressrace PKG=./pkg/ccl/partitionccl TESTS=(TestInitialPartitioning|TestRepartitioning) TESTTIMEOUT=3m45s STRESSFLAGS=-stderr -maxfails 1 -maxtime 5m0s`. These tests have many sub-tests and run longer than 3m45s.

petermattis committed:
ccl/partitionccl: workaround stressrace failures

Skip `Test{InitialPartition,Repartitioning}` for nightly stressrace
runs. These tests configure many sub-tests and hit the configured time limit.

Release note: None

petermattis commented:
Ah, this is coming from `github-pull-request-make`. I've added another commit which skips both tests for nightly stressrace runs. Perhaps it would be better to tweak the timeouts in `github-pull-request-make`. PTAL.

petermattis commented:
bors r=benesch

benesch commented:
LGTM. Looks like Examples-ORMs flaked so queuing you a new build.

bors r+

> Perhaps it would be better to tweak the timeouts in github-pull-request-make.

As you might have discovered, that's complicated by the fact that if you touch multiple packages gprm will use a timeout as low as 45s. I think what you did here is the best available option.

petermattis commented:
`Examples-ORMs` flake (again).

bors r=benesch

danhhz commented:
Huh. This is a pretty straightforward fix. The issue I was seeing the last time I looked into this (https://github.com/cockroachdb/cockroach/issues/28789#issuecomment-428004355) was a deadlock in pkg/storage, which is why I punted it over to core. Perhaps whatever that issue was has been fixed for some other reason since.
